ipfsconn/ipfshttp/ipfshttp.go

```diff
@@ -713,14 +721,26 @@ func (ipfs *Connector) PinLsCid(hash *cid.Cid) (api.IPFSPinStatus, error) {
 	return api.IPFSPinStatusFromString(pinObj.Type), nil
 }
 
-func doPost(client *http.Client, apiURL, path string) (*http.Response, error) {
+func (ipfs *Connector) doPost(client *http.Client, apiURL, path string) (*http.Response, error) {
 	logger.Debugf("posting %s", path)
-	url := fmt.Sprintf("%s/%s", apiURL, path)
+	urlstr := fmt.Sprintf("%s/%s", apiURL, path)
 
-	res, err := client.Post(url, "", nil)
+	ctx, cancel := context.WithCancel(ipfs.ctx)
```

Comment:
Yeah, there was a reason but I have forgotten it and also I have learnt a lot more about the `context` pkg since I wrote this, so my guess is that my reason isn't necessarily valid anymore anyway